scriptmodules/supplementary/bluetooth.sh

```diff
@@ -300,7 +320,7 @@ function boot_bluetooth() {
             while read mac_address; read device_name; do
                 macs+=($mac_address)
             done < <(list_registered_bluetooth)
-            local script="while true; do for mac in ${macs[@]}; do hcitool con | grep -q \"\$mac\" || echo \"connect \$mac\" | bluetoothctl >/dev/null 2>&1; sleep 10; done; done"
+            local script="while true; do for mac in ${macs[@]}; do hcitool con | grep -q \"\$mac\" || bluez_cmd \"connect \$mac\"; sleep 10; done; done"
```

Comment:
I don't think this will work as this script is backgrounded. the bluez_cmd won't be available.

Reply:
The issue with the new bluez is that its command-line parsing no longer works with a direct pipe unless you pad with an explicit sleep before and after the required command, i.e. something like:

```{ sleep 1; echo \"connect \$mac\"; sleep 1; } | bluetoothctl >/dev/null 2>&1```

Would something like the above be a satisfactory workaround to using the dedicated function? I'm not sure if I'm missing more escape characters for the curly braces etc., but I'll test.

Reply:
Also, the original command is also missing the "quit" command, which may leave a rogue bluetoothctl process.

Reply:
I never noticed any bluetoothctl processes being left when testing - it always quit after the pipe closed. So the bluetoothctl command line parsing got even worse :-) nice. If your workaround works, I'm happy to switch to it.

Reply:
Well - testing bluez 5.43 on stretch:
```
{ echo version; sleep 1; } | bluetoothctl
```
gives: ```(bluetoothctl:16701): GLib-CRITICAL **: Source ID 23 was not found when attempting to remove it``` - but the process is gone, yes.

```
{ echo -e "version\nquit"; sleep 1; } | bluetoothctl
```
Will quit gracefully. Probably a good idea to do it.

My memory is foggy on whether sleep needs to happen first, but it's probably a good idea since the dedicated function properly waits for the correct "bluetooth#" prompt to appear, but a direct pipe wouldn't.

Reply:
I agree. I don't remember this on Jessie, so maybe it's also new.

Reply:
Didn't seem to need the sleep here at least for the version command.

Reply:
Some commands are ok, including ```version``` and ``ìnfo ...``` but the "connect" command was failing for me on BlueZ 5.48. 

The Raspbian patches to BlueZ change the default dbus security policy to more permissive, so perhaps the behaviour was different in an older version of the PR where I hadn't yet included the patches. Will re-check it regardless.